tests/tests_core/test_json_serializer.py

```diff
@@ -0,0 +1,86 @@
+        statistical_test_result = StatisticalTestResult(statistical_test, simple_stats)
+
+        js_result = statistical_test_result.toJson()  # no error/exception should be raise
+        print(js_result)    # use pytest -s to check the output if needed
+
+    def test_serialize_multi_test_suite_result(self):
+        control_statistics   = SampleStatistics(1000, 0.0, 1.0)
+        treatment_statistics = SampleStatistics(1200, 1.0, 1.0)
+        delta                = 1.0
+        p                    = 0.04
+        statistical_power    = 0.8
+        confidence_interval  = ConfidenceInterval({2.5: 0.1, 97.5: 1.1})
+        stop                 = True
+
+        corrected_p          = 0.02
+        corrected_ci         = ConfidenceInterval({1.0: -0.7, 99.0: 0.7})
+        corrected_stop       = False
+
+        simple_stats = SimpleTestStatistics(control_statistics,
+                                            treatment_statistics,
+                                            delta,
+                                            confidence_interval,
+                                            p,
+                                            statistical_power)
+
+        simple_stats_corrected = SimpleTestStatistics(control_statistics,
+                                                      treatment_statistics,
+                                                      delta,
+                                                      corrected_ci,
+                                                      corrected_p,
+                                                      statistical_power)
+
+        es_stats = EarlyStoppingTestStatistics(control_statistics,
+                                               treatment_statistics,
+                                               delta,
+                                               confidence_interval,
+                                               p,
+                                               statistical_power,
+                                               stop)
+
+        es_stats_corrected = EarlyStoppingTestStatistics(control_statistics,
+                                                         treatment_statistics,
+                                                         delta,
+                                                         corrected_ci,
+                                                         corrected_p,
+                                                         statistical_power,
+                                                         corrected_stop)
+
+
+        test_results = [CorrectedTestStatistics(simple_stats, simple_stats_corrected),
+                        CorrectedTestStatistics(es_stats, es_stats_corrected)]
+        statistical_test_results = MultipleTestSuiteResult(test_results, MultipleTestingCorrectionMethod.no_correction)
+
+        js_result = statistical_test_results.toJson()  # no error/exception should be raise
```

Comment:
This certainly doesn't throw an exception/error, but do all fields of `statistical_test_results` get properly serialized in a recursive manner?

Reply:
yes it works recursively because of the following:

```
class JsonSerializable(object):
    def toJson(self):
        return json.dumps(self, default=lambda o: o.name if isinstance(o, Enum) else o.__dict__, sort_keys=True, indent=4)
```

It actually can only serialize at a flat level and throw an error saying it can't be serialzed without the above lambda.